addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScannerUnitTest.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.Base64;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class JsoScannerUnitTest extends PassiveScannerTest<JsoScanner> {
+
+    public static final String URI_ENCODED_JSO =
```

Comment:
This could be generated at runtime, it would be one less thing to change when the scanner is promoted.

Reply:
I would like to do it this way but I have not found a way to properly convert the JSO byte array to this equivalent URI form.
I would like some help on this part.

Reply:
`URLEncoder.encode(new String(createJso(), StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8.name())` should give the same result.

Reply:
That works! I should take some time to understand why...
Anyway, Thanks! Hum, I still have a lot of things to learn -_-'